docs/migrating.rst

```diff
@@ -1095,6 +1153,55 @@ The following changes have been made:
 Command Extension Changes
 ---------------------------
 
+.. _migrating_2_0_commands_extension_cog_async:
+
+Extension and Cog Loading / Unloading is Now Asynchronous
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As an extension to :ref:`migrating_2_0_client_async_setup` changes the loading and unloading of extensions and cogs is now asynchronous.
+
+To accommodate this, the following changes have been made:
+
+- the ``setup`` and ``teardown`` functions in extensions must now be coroutines.
+- :meth:`ext.commands.Bot.load_extension` must now be awaited.
+- :meth:`ext.commands.Bot.unload_extension` must now be awaited.
+- :meth:`ext.commands.Bot.reload_extension` must now be awaited.
+- :meth:`ext.commands.Bot.add_cog` must now be awaited.
+- :meth:`ext.commands.Bot.remove_cog` must now be awaited.
+
+Quick example of an extension setup function:
+
+.. code:: python
+
+    # before
+    def setup(bot):
+        bot.add_cog(MyCog(bot))
+
+    #after
+    async def setup(bot):
+        await bot.add_cog(MyCog(bot))
+
+Quick example of loading an extension:
+
+.. code:: python
+
+    #before
+    bot.load_extension('my_extension')
+
+    #after using setup_hook
+    @bot.event
+    async def setup_hook():
+        await bot.load_extension('my_extension')
```

Comment:
We're mostly moving away from the free form decorators for a while now so it's best to not recommend this.